config/initializers/fog.rb

```diff
@@ -0,0 +1,27 @@
+require 'rubygems'
+require 'fog'
+
+CarrierWave.configure do |config|
+  # Configuration for Amazon S3
+  config.fog_credentials = {
+      provider: 'AWS',
+      aws_access_key_id: "#{Rails.application.secrets.sorcery_s3_key}",
+      aws_secret_access_key: "#{Rails.application.secrets.sorcery_s3_secret}",
+      region: ENV['S3_REGION']
+  }
+
+  # For testing, upload files to local `tmp` folder.
+  if Rails.env.test? || Rails.env.cucumber?
+    config.storage = :file
+    config.enable_processing = false
+    config.root = "#{Rails.root}/tmp"
+  else
+    config.storage = :fog
+  end
+
+  # To let CarrierWave work on heroku
+  config.cache_dir = "#{Rails.root}/public/uploads/tmp"
+
+  config.fog_directory = ENV['S3_BUCKET_NAME']
+  config.fog_attributes = { 'Cache-Control' => 'max-age=315576000'}
```

Comment:
Space inside } missing.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.